overlord/snapstate/handlers_components.go

```diff
@@ -248,6 +248,12 @@ func (m *SnapManager) doMountComponent(t *state.Task, _ *tomb.Tomb) (err error)
 	// TODO we might want a checkComponents doing checks for some
 	// component types (see checkSnap and checkSnapCallbacks slice)
 
+	if err := ensureSnapAndComponentsAssertionStatus(
+		*snapsup.SideInfo, []snap.ComponentSideInfo{*compSetup.CompSideInfo},
+	); err != nil {
+		return err
+	}
+
```

Comment:
Could this be done in `doInstallComponent`, before the change is created?

Reply:
As discussed, if this is possibly a double check and mainly about being on the safe side, maybe it would be worth adding a comment.